sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -621,12 +621,23 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
       ds.filter(_ => true),
       Some(1), Some(2), Some(3))
   }
+
+  test("SPARK-13540 add Scala object to OuterScopes") {
+    OuterScopes.addOuterScope(OuterObject)
```

Comment:
I'm a little confused here. If nested classes defined in scala object don't need out scope, why do we add out scope in this test case?